Can't build for macOS, Pods-Runner does not support provisioning profiles, but provisioning profile match Development * has been manually specified

hot_mode_dev_cycle_macos_target__benchmark is broken currently and complaining of a cert issue

https://flutter-dashboard.appspot.com/#/build

Added verbose logging in https://github.com/flutter/flutter/pull/71610 to track down the issues.

```
Pods-Runner does not support provisioning profiles, but provisioning profile match Development * has been manually specified.
```

Is this just a config issue, rather than cert problem?

/Users/flutter/.cocoon/flutter/dev/integration_tests/flutter_gallery$ ../../../bin/flutter build macos
note: Using new build system                                            
note: Planning build                                                    
note: Constructing build description                                    
error: Pods-Runner does not support provisioning profiles. Pods-Runner does not support provisioning profiles, but provisioning profile match Development * has been manually specified. Set the provisioning profile value to "Automatic" in the build settings editor. (in target 'Pods-Runner' from project 'Pods')
** BUILD FAILED **                                                      
                                                                        
Building macOS application...                                           
Build process failed

--- a/packages/flutter_tools/bin/podhelper.rb
+++ b/packages/flutter_tools/bin/podhelper.rb
@@ -115,6 +115,12 @@ def flutter_additional_macos_build_settings(target)
     # When deleted, the deployment version will inherit from the higher version derived from the 'Runner' target.
     # If the pod only supports a higher version, do not delete to correctly produce an error.
     build_configuration.build_settings.delete 'MACOSX_DEPLOYMENT_TARGET' if inherit_deployment_target
+
+    # Avoid error about Pods-Runner not supporting provisioning profiles.
+    # Framework signing is handled at the app layer, not per framework, so disallow individual signing.
+    build_configuration.build_settings.delete 'EXPANDED_CODE_SIGN_IDENTITY'
+    build_configuration.build_settings['CODE_SIGNING_REQUIRED'] = 'NO'
+    build_configuration.build_settings['CODE_SIGNING_ALLOWED'] = 'NO'
   end
 end
 

